cmd/antrea-agent/agent.go

```diff
@@ -184,11 +184,11 @@ func run(o *Options) error {
 		v6Enabled := config.IsIPv6Enabled(nodeConfig, networkConfig.TrafficEncapMode)
 		switch {
 		case v4Enabled && v6Enabled:
-			proxier = proxy.NewDualStackProxier(nodeConfig.Name, informerFactory, ofClient)
+			proxier = proxy.NewDualStackProxier(nodeConfig.Name, nodeConfig.GatewayConfig, informerFactory, ofClient, routeClient)
 		case v4Enabled:
-			proxier = proxy.NewProxier(nodeConfig.Name, informerFactory, ofClient, false)
+			proxier = proxy.NewProxier(nodeConfig.Name, nodeConfig.GatewayConfig, informerFactory, ofClient, routeClient, false)
```

Comment:
Since nodeConfig.GatewayConfig already includes IPv4 and/or IPv6 configurations, do you think we could remove the last parameter "isIPv6", and check if nodeConfig.GatewayConfig is configuring with IPv4 or/and IPv6?

Reply:
Maybe no.

The first reason is that it depends on the results of `config.IsIPv4Enabled()` and `config.IsIPv6Enabled()` if there's an IPv4/IPv6 proxier. Such condition doesn't equal to if IPv4/IPv6 is nil in `nodeConfig.GatewayConfig`.

The second reason is that, considering implementation, there will always be a function to deal with one specific proxier. If dual-stack then run it twice and combine the result. For such function, a parameter of IP family is needed. So even if I remove the `isIPv6` paramter here, I need to decide IP family within `proxy.NewProxier()` anyway.